### ISSUE TITLE
cookiecutter does not cleanup files if panics during template expansion

### ISSUE THREAD

**Issue by MisterEggnog** [cookiecutter/cookiecutter#1754]
* Cookiecutter version: 2.1.1
* Template project url: Zip file provided
* Python version: 3.8.10
* Operating System: Ubuntu Linux

### Description:

While working on a cookiecutter template I added a variable to cookiecutter.json that had a dash in the name.
This failed, which is understandable, however, I found that the output was not cleaned up after the failure.
It appears that the cause of this is that cookiecutter itself is crashing, before it can clean up files.

### What I've run:
Input:
```
cookiecutter failing-cookie-cutter.zip
normal [normal]: 
word-with_dash [this fails]: 
```

Program output
```
Traceback (most recent call last):
  File "/home/gsinis/.local/bin/cookiecutter", line 8, in <module>
    sys.exit(main())
  File "/usr/lib/python3/dist-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/gsinis/.local/lib/python3.8/site-packages/cookiecutter/cli.py", line 194, in main
    cookiecutter(
  File "/home/gsinis/.local/lib/python3.8/site-packages/cookiecutter/main.py", line 114, in cookiecutter
    result = generate_files(
  File "/home/gsinis/.local/lib/python3.8/site-packages/cookiecutter/generate.py", line 383, in generate_files
    _run_hook_from_repo_dir(
  File "/home/gsinis/.local/lib/python3.8/site-packages/cookiecutter/generate.py", line 252, in _run_hook_from_repo_dir
    run_hook(hook_name, project_dir, context)
  File "/home/gsinis/.local/lib/python3.8/site-packages/cookiecutter/hooks.py", line 131, in run_hook
    run_script_with_context(script, project_dir, context)
  File "/home/gsinis/.local/lib/python3.8/site-packages/cookiecutter/hooks.py", line 111, in run_script_with_context
    output = template.render(**context)
  File "/home/gsinis/.local/lib/python3.8/site-packages/jinja2/environment.py", line 1291, in render
    self.environment.handle_exception()
  File "/home/gsinis/.local/lib/python3.8/site-packages/jinja2/environment.py", line 926, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "<template>", line 1, in top-level template code
jinja2.exceptions.UndefinedError: 'collections.OrderedDict object' has no attribute 'word'
```
Output using ls
```
normal
```
[failing-cookie-cutter.zip](https://github.com/cookiecutter/cookiecutter/files/9185761/failing-cookie-cutter.zip)

**Comment by liortct** [comment]
I fixed this bug in the PR #1760 .
I advice to merge the PR and close this issue.

**Comment by liortct** [comment]
Hi!
We started working on a new community fork of the Cookiecutter project, named Cookieninja.
The fork is found at: https://github.com/cookieninja-generator/cookieninja

We have re-opened your issue at https://github.com/cookieninja-generator/cookieninja/issues/27

We're trying to bring the life back to this amazing project.
We are committed to addressing each issue in up to 1 week from the day it's opened.
We'd really like you to join the fork, star it and contribute to it!

Thank you :)

**Comment by kurtmckee** [comment]
Resolved by #1760.
